src/agora/dobeamweights.cc

```diff
@@ -376,8 +440,9 @@ static inline void PartialTransposeGather(size_t cur_sc_id, float* src,
 
 // Gather data of one symbol from partially-transposed buffer
 // produced by dofft
-static inline void TransposeGather(size_t cur_sc_id, float* src, float*& dst,
-                                   size_t bs_ant_num, size_t ofdm_data_num) {
+inline void DoBeamWeights::TransposeGather(size_t cur_sc_id, float* src,
```

Comment:
Not sure if a &*  is really necessary.  Check the inputs and outputs and see if you can simplify the types